libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryUtils.java

```diff
@@ -173,6 +178,37 @@ public static String generateCreateDateFormatted(Date date) {
     return dateFormat.format(date);
   }
 
+  /**
+   * This method retrieves metadata.
+   *
+   * @param context application context
+   * @return metadata
+   */
+  @Nullable
+  public static Bundle obtainMetaData(Context context) {
+    try {
+      return context.getPackageManager().getApplicationInfo(context.getPackageName(),
+        PackageManager.GET_META_DATA).metaData;
+    } catch (Exception exception) {
+      Log.e(TAG, exception.getMessage());
+      return null;
+    }
+  }
+
+  public static int obtainInitDelay(Context context) {
+    int initDelay = DEFAULT_INIT_DELAY;
+    Bundle metadata = obtainMetaData(context);
+
+    if (metadata != null) {
+      int configuredDelay = metadata.getInt(KEY_META_DATA_INIT_DELAY);
+      if (configuredDelay != 0 && configuredDelay <= MAX_INIT_DELAY) {
```

Comment:
Use `configuredDelay > 0` here. If a minus value is passed in, the delay function will not works.